samples/nrf9160/lte_ble_gateway/prj.conf

```diff
@@ -19,6 +19,7 @@ CONFIG_MQTT_MAX_PACKET_LENGTH=2048
 
 # LTE link control
 CONFIG_LTE_LINK_CONTROL=y
+CONFIG_LTE_AUTO_INIT_AND_CONNECT=n
```

Comment:
I agree that another PR could do some changes to `lte_lc_init_and_connect` to avoid calling it if the link is already up, but not sure about doing it compile-time, as it will be ok if for example `lte_lc_offline` is called between auto-connect and calling `lte_lc_init_and_connect`. Calling it when the link is up wouldn't do any harm to the modem/link I think, but the function would hang forever in a while loop waiting for a specific response that will never come.